js/FilterBarComponent.js

```diff
@@ -456,12 +457,20 @@ function populateFilterList ($filterbar) {
         if (filterValue !== '' && filterValue !== null) {
             $labelContainer.prepend('<span class="label label--large label--inverse" data-filter-id="' + filterId + '">' + filterLabel + filterValue + '<a data-ui="filter-cancel" class="btn remove-button" data-filter-id="'+ filterId +'"><i class="icon-remove-sign"></i></a></span>');
 
+            // Keep track of how many filters have already been applied
+            hiddenFormGroups++;
+
             // Hide the filter item in the list
             updateFilterList($addFilterButton, filterId, 'hide');
 
             // Show the filterbar
             $filterbar.removeClass('display--none');
         }
+
+        // Don't show add filter button if all filters have been loaded
+        if ($formGroups.length === hiddenFormGroups) {
+            $addFilterButton.addClass('display--none');
```

Comment:
Should we not be using `u-display--none` here instead to try move towards the newer utility classes.